src/terms/anyonic.jl

```diff
@@ -2,14 +2,60 @@
 # This term does not contain the kinetic energy, which must be added separately
 # /!\ They have no 1/2 factor in front of the kinetic energy,
 #     so for consistency the added kinetic energy must have a scaling of 2
-# Energy = <u, ((-i∇ + βA)^2 + V) u>
+# Energy = <u, ((-i hbar ∇ + βA)^2 + V) u>
 # where ∇∧A = 2π ρ, ∇⋅A = 0 => A = x^⟂/|x|² ∗ ρ
-# H = (-i∇ + βA)² + V - 2β x^⟂/|x|² ∗ (βAρ + J)
-#   = -Δ + 2β (-i∇)⋅A + β²|A|^2 - 2β x^⟂/|x|² ∗ (βAρ + J)
+# H = (-i hbar ∇ + βA)² + V - 2β x^⟂/|x|² ∗ (βAρ + J)
+#   = - hbar^2 Δ + 2β hbar (-i∇)⋅A + β²|A|^2 - 2β x^⟂/|x|² ∗ (βAρ + J)
 # where only the first three terms "count for the energy", and where
-# J = 1/(2i) (u* ∇u - u ∇u*)
+# J = 1/(2i) hbar (u* ∇u - u ∇u*)
+
+# for numerical reasons, we solve the equation ∇∧A = 2π ρ in two parts: A = A_SR + A_LR, with
+# ∇∧A_SR = 2π(ρ-ρref)
+# ∇∧A_LR = 2πρref
+# where ρref is a gaussian centered at the origin and with the same integral as ρ (=M)
+# ρref(x) = M exp(-1/2 (x/σ)^2) / (σ^3 (2π)^3/2)
+# This way, ρ-ρref has zero mass, and A_SR is shorter range. The long-range part is computed explicitly
+
+function ρref_real(x::T, y::T, M, σ) where {T <: Real}
+    r = hypot(x, y)
+    M * exp(-1/2 * (r/σ)^2) / (σ^2 * (2T(π))^(2/2))
```

Comment:
The `2/2` is a bit strange. I think I would assign `dim = 2` and than use that variable to make that clear. Also be careful with type stability here `1/2` will always be Float64.

Reply:
it's just the formula for a normalized gaussian, assigning dim = 2 would just be a waste of space I think...

Reply:
Come on it's one line!